x/rollup/types/expected_keepers.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BankKeeper defines the expected bank keeper interface used in the x/rollup module
+type BankKeeper interface {
```

Comment:
Are there reasons to put the interface in the `types` package instead of in the `keeper` package? Is this an SDK convention?

Reply:
It's just a Cosmos SDK quirk. It's mentioned as being in the types package in the [Folder Structure docs](https://docs.cosmos.network/main/build/building-modules/structure#structure).

The [keeper docs](https://docs.cosmos.network/main/build/building-modules/keeper#type-definition) touch on it a little more:
> An expected keeper is a keeper external to a module that is required by the internal keeper of said module. External keepers are listed in the internal keeper's type definition as interfaces. These interfaces are themselves defined in an expected_keepers.go file in the root of the module's folder. In this context, interfaces are used to reduce the number of dependencies, as well as to facilitate the maintenance of the module itself.

Reply:
Great reference, thanks. Not a fan of the convention myself, but probably best to follow it here.